File: mat/max_ok.m
function [ok] = max_ok()
ok  = true;
m   = 2;
n   = 3;
x   = rand(m, n);
% ------------------
y   = max( max(x) );
% ------------------
[m, n] = size(y);
ok     = ok & (m == 1) & (n == 1); 
ok     = ok & all( all ( x <= y ) );
ok     = ok & ~ all( all ( x < y ) );
return

File: cpp/max_ok.cpp
# include <mat2cpp.hpp>
bool max_ok(void)
{	bool   ok  = true;
	using namespace mat2cpp;

	size_t i, j, m(2), n(3);
	matrix<double> x = rand(m, n);
	// ---------------
	double y = max(x);
	// ---------------
	bool less = true;
	for(i = 0; i < m; i++)
	{	for(j = 0; j < n; j++)
		{	ok   &= x(i,j) <= y;
			less &= x(i,j) < y;
		}
	}
	ok &= ! less;
	return ok;
}
